iothub/device/devdoc/API Design.md

```diff
@@ -14,72 +15,72 @@ public Task<Properties> GetPropertiesAsync(CancellationToken cancellationToken =
-/// <param name="conventionHandler">A convention handler that defines the content encoding and serializer to use for the properties.</param>
+/// <param name="propertyConvention">A convention handler that defines serializer to use for the properties.</param>
+/// <param name="componentName">The component name this property belongs to.</param>
 /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
-public Task UpdatePropertiesAsync(IDictionary<string, object> properties, string componentName = default, IConventionHandler conventionHandler = default, CancellationToken cancellationToken = default);
+public Task UpdatePropertiesAsync(IDictionary<string, object> properties, PropertyConvention propertyConvention, string componentName = default, CancellationToken cancellationToken = default);
 
 /// <summary>
-/// Sets the global listener for Writable properties
+/// Update a writable property.
 /// </summary>
-/// <param name="callback">The global call back to handle all writable property updates.</param>
-/// <param name="userContext">Generic parameter to be interpreted by the client code.</param>
-/// <param name="cancellationToken">A cancellation token.</param>
-public Task SubscribeToWritablePropertyEvent(Action<Properties, object> callback, object userContext, CancellationToken cancellationToken = default);
+/// <param name="propertyName">Property name.</param>
+/// <param name="writablePropertyResponse">The writable properyt response to push.</param>
+/// <param name="componentName">The component name this property belongs to.</param>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+public Task UpdateWritablePropertyAsync(string propertyName, WritablePropertyResponse writablePropertyResponse, string componentName = default, CancellationToken cancellationToken = default);
 
 /// <summary>
 /// Sets the global listener for Writable properties
 /// </summary>
 /// <param name="callback">The global call back to handle all writable property updates.</param>
-/// <param name="componentName">The component name this writable property belongs to.</param>
 /// <param name="userContext">Generic parameter to be interpreted by the client code.</param>
-/// <param name="cancellationToken">A cancellation token.</param>
-public Task SubscribeToWritablePropertyEvent(Action<Properties, object> callback, string componentName = default, object userContext = default, CancellationToken cancellationToken = default);
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+public Task SubscribeToWritablePropertyEventAsync(Func<PropertyCollection, object, Task> callback, object userContext, CancellationToken cancellationToken = default)
 ```
 
 <details>
 <summary>Properties</summary>
 
 ```csharp
 /// <summary>
-/// A container for properties for your device
+/// A container for properties.
 /// </summary>
 public class Properties
```

Comment:
naming HubProperties?

Reply:
DeviceProperties I beleive

Reply:
ClientProperties (can't put "device" in there)